src/dawn/Support/ASTSerialier.cpp

```diff
@@ -0,0 +1,673 @@
+  expr->getLeft()->accept(*this);
+  currentExprProto_.pop();
+
+  currentExprProto_.push(protoExpr->mutable_right());
+  expr->getRight()->accept(*this);
+  currentExprProto_.pop();
+
+  setLocation(protoExpr->mutable_loc(), expr->getSourceLocation());
+}
+
+void ProtoStmtBuilder::visit(const std::shared_ptr<TernaryOperator>& expr) {
+  auto protoExpr = getCurrentExprProto()->mutable_ternary_operator();
+
+  currentExprProto_.push(protoExpr->mutable_cond());
+  expr->getCondition()->accept(*this);
+  currentExprProto_.pop();
+
+  currentExprProto_.push(protoExpr->mutable_left());
+  expr->getLeft()->accept(*this);
+  currentExprProto_.pop();
+
+  currentExprProto_.push(protoExpr->mutable_right());
+  expr->getRight()->accept(*this);
+  currentExprProto_.pop();
+
+  setLocation(protoExpr->mutable_loc(), expr->getSourceLocation());
+}
+
+void ProtoStmtBuilder::visit(const std::shared_ptr<FunCallExpr>& expr) {
+  auto protoExpr = getCurrentExprProto()->mutable_fun_call_expr();
+  protoExpr->set_callee(expr->getCallee());
+
+  for(const auto& arg : expr->getArguments()) {
+    currentExprProto_.push(protoExpr->add_arguments());
+    arg->accept(*this);
+    currentExprProto_.pop();
+  }
+
+  setLocation(protoExpr->mutable_loc(), expr->getSourceLocation());
+}
+
+void ProtoStmtBuilder::visit(const std::shared_ptr<StencilFunCallExpr>& expr) {
+  auto protoExpr = getCurrentExprProto()->mutable_stencil_fun_call_expr();
+  protoExpr->set_callee(expr->getCallee());
+
+  for(const auto& arg : expr->getArguments()) {
+    currentExprProto_.push(protoExpr->add_arguments());
+    arg->accept(*this);
+    currentExprProto_.pop();
+  }
+
+  setLocation(protoExpr->mutable_loc(), expr->getSourceLocation());
+}
+
+void ProtoStmtBuilder::visit(const std::shared_ptr<StencilFunArgExpr>& expr) {
+  auto protoExpr = getCurrentExprProto()->mutable_stencil_fun_arg_expr();
+
+  protoExpr->mutable_dimension()->set_direction(
+      expr->getDimension() == -1
+          ? dawn::proto::statements::Dimension::Invalid
+          : static_cast<dawn::proto::statements::Dimension_Direction>(expr->getDimension()));
+  protoExpr->set_offset(expr->getOffset());
+  protoExpr->set_argument_index(expr->getArgumentIndex());
+
+  setLocation(protoExpr->mutable_loc(), expr->getSourceLocation());
+}
+
+void ProtoStmtBuilder::visit(const std::shared_ptr<VarAccessExpr>& expr) {
+  auto protoExpr = getCurrentExprProto()->mutable_var_access_expr();
+
+  protoExpr->set_name(expr->getName());
+  protoExpr->set_is_external(expr->isExternal());
+
+  if(expr->isArrayAccess()) {
+    currentExprProto_.push(protoExpr->mutable_index());
+    expr->getIndex()->accept(*this);
+    currentExprProto_.pop();
+  }
+
+  setLocation(protoExpr->mutable_loc(), expr->getSourceLocation());
+}
+
+void ProtoStmtBuilder::visit(const std::shared_ptr<FieldAccessExpr>& expr) {
+  auto protoExpr = getCurrentExprProto()->mutable_field_access_expr();
+
+  protoExpr->set_name(expr->getName());
```

Comment:
In the case of stencil function inline example you passed me of horizontal diffusion, 
the names stores are the names of the stencil function scope, while we need the original names. 
There is something wrong here

Reply:
I'll investigate